softserve/lib.py

```diff
@@ -36,14 +36,21 @@ def wrap(*args, **kwargs):
 
 
 @celery.task()
-def create_node(counts, name, node_request, pubkey):
+def create_node(counts, name, os_type, node_request, pubkey):
     '''
     Create a node in the cloud provider
     '''
     conn = boto.ec2.connect_to_region(app.config['REGION_NAME'],
         aws_access_key_id=app.config['AWS_ACCESS_KEY_ID'],
         aws_secret_access_key=app.config['AWS_SECRET_ACCESS_KEY'])
     conn.import_key_pair(name, pubkey)
+
+    #Deciding the OS_TYPE
+    if os_type == 'Centos-7':
+        IMAGE_ID = IMAGE_ID_7
+    else:
+        IMAGE_ID = IMAGE_ID_8
```

Comment:
While calling a run_instance function, it still uses app.config['IMAGE_ID']. Your logic won't work well here.